src/main/java/org/black_ixx/bossshop/core/rewards/BSRewardTypePlayerCommandOp.java

```diff
@@ -43,9 +43,14 @@ public void giveReward(Player p, BSBuy buy, Object reward, ClickType clickType)
         if (p.isOp()) {
             executeCommands(p, buy, commands);
         } else {
-            p.setOp(true);
-            executeCommands(p, buy, commands);
-            p.setOp(false);
+            try {
+                p.setOp(true);
+                executeCommands(p, buy, commands);
+                p.setOp(false);
+            } catch (Exception e) {
+                e.printStackTrace();
+                p.setOp(false);
```

Comment:
Nice catch. Let's set the OP to false in a `finally` block and add just a little bit if logging to the catch to explain what happened.